okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/HostMetrics.java

```diff
@@ -36,6 +36,11 @@
      */
     String hostname();
 
+    /**
+     * The url used to gather the metrics.
+     */
+    String url();
```

Comment:
where is this used?

Reply:
In the health checks for remote services for witchcraft. We currently use the hostname rather than the url.

Reply:
Downstream consumers would be parsing this and could handle the unsafe params. Although to avoid this altogether, we could reconstruct the uri via the `scheme`, `hostname` and `port`, which would be enough to uniquely identify it without the full url

Reply:
we can add the port if it's critical for disambiguation. not a fan of the "parse later" proposal, it's very brittle.

Reply:
yeah, curious what this information will be used for?